File: JAVA/binary_search.java
import java.util.Scanner;

class Main {
  int binarySearch(int array[], int element, int low, int high) {
    while (low <= high) {

      int mid = low + (high - low) / 2;

      if (array[mid] == element)
        return mid;


      if (array[mid] < element)
        low = mid + 1;


      else
        high = mid - 1;
    }

    return -1;
  }

  public static void main(String args[]) {


    Main obj = new Main();


    int[] array = { 3, 4, 5, 6, 7, 8, 9 };
    int n = array.length;

    Scanner input = new Scanner(System.in);

    System.out.println("Enter element to be searched:");

    int element = input.nextInt();
    input.close();


    int result = obj.binarySearch(array, element, 0, n - 1);
    if (result == -1)
      System.out.println("Not found");
    else
      System.out.println("Element found at index " + result);
  }
}

File: Python/Binary_Search.py
def binary_search(my_arr, low, high, x):
 

    if high >= low:
 
        mid = (high + low) // 2
 
        if my_arr[mid] == x:
            return mid
 

        elif my_arr[mid] > x:
            return binary_search(my_arr, low, mid - 1, x)

        else:
            return binary_search(my_arr, mid + 1, high, x)
 
    else:

        return -1
 

my_arr = [ 2, 3, 4, 10, 40 ]
x = 2
 

result = binary_search(my_arr, 0, len(my_arr)-1, x)
 
if result != -1:
    print("Element is present at index", str(result))
else:
    print("Element is not present in array")